File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.java
import java.util.*;

class Solution {
    public int solution(int[] nums) {
        HashSet<Integer> set = new HashSet<>();
        int answer = 0;
        
        for (int n : nums) {
            set.add(n);
        }
        
        int c = (nums.length) / 2;
        int s = set.size(); 

        return (c > s)? s : c;
    }
}

File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.cpp
#include <vector>
#include <unordered_set>

using namespace std;

int solution(vector<int> nums)
{
    unordered_set<int> hash_set; 
    
    for (int n : nums) {
        hash_set.insert(n);
    }
    
    int select = nums.size() / 2;
    int count = hash_set.size();
        
    return (select > count)? count : select;
}